### ISSUE TITLE
script "text/template" tag and LEGACYHTML5 mode

### ISSUE THREAD

**Issue by sebdavid** [thymeleaf/thymeleaf#262]
Hello,

I'm using Thymeleaf 2.1.2. It seems that the following code works in HTML5 mode, but not in LEGACYHTML5 :

``` html
<script type="text/template" id="myTemplate" th:fragment="waiter">
    <div>
        <span th:text="#{LOADING}">Chargement...</span>
        <img th:src="@{/resources/img/ajax_process.gif}" />
    </div>
</script>
```

HTML5 mode : the "th:" attributes are well processed
LEGACYHTML5 : the "th:" attributes are not processed, and are displayed as they are written.

Sébastien.

**Comment by danielfernandez** [comment]
This has been verified. Unfortunately, this will be quite difficult to solve in 2.1.

The problem is that **nekoHTML**, which is used for parsing in `LEGACYHTML5` mode, always treats the contents of a `<script>` tag as text not markup so, after parsing this fragment, the whole content of the `<script>` is fed to the Thymeleaf engine a `Text` DOM node. This makes it impossible for thymeleaf to execute attribute processors the _usual_ way. 

The workaround is, therefore, to use _text inlining_ in the way exemplified in this test: https://github.com/thymeleaf/thymeleaf-tests/blob/20140502/src/test-21/resources/engine21/parsing/parsing22.thtest

**Comment by danielfernandez** [comment]
The `LEGACYHTML5` template mode has dissapeared from Thymeleaf 3.0 and `<script>` tags are considered non-processable CDATA now. In this case, a `<template>` tag should be used, and that should work just fine. See #390 

Already in `3.0.0-SNAPSHOT`

**Comment by danielfernandez** [comment]
Please note this has been modified.

Given the lack of support for `<template>` in some major browsers like Internet Explorer 11, I've implemented attoparser/attoparser#8 so that `<script type="text/template">` tags are **not** considered CDATA and therefore their bodies will be parsed in Thymeleaf 3.0 (contrary to any other `<script>` element, which will still be CDATA).

This corrected behaviour has already been implemented into `3.0.0-SNAPSHOT`
